### PR TITLE
test suites for gitbase + bblfshd(mocked up)

### LINKED ISSUE #18: Research: gitbase + bblfshd mockups
Need to check a possibility of implementation of `bblfshd` mockups to test gitbase's ability to handle `bblfshd` failures long evaluations etc.

### PR DESCRIPTION
**Pull request by lwsanty**
This is a suite that checks if gitbase correctly handles the difficulties that may appear during the interaction with bblfshd GRPC server, such as parse request errors, parse errors, broken UASTs, long parse time evaluation etc.

It uses different bblfshd mockups to be served, regression-core to spin up the gitbase and perform queries.

To assert gitbase behavior it's stderr is being parsed after the query execution.

This tests are been compiled to a simple binary that's added to regression gitbase docker and should be launched during the gitbase regression test routine.

closes #18

Signed-off-by: lwsanty <lwsanty@gmail.com>

### CODE REVIEW

**Comment by lwsanty** [comment]
this pr is blocked by https://github.com/src-d/gitbase/issues/950

**Comment by jfontan** [comment]
Maybe I understand the changes incorrectly but these tests seem like integration tests and I feel they belong to gitbase repo instead of regression.

**Comment by lwsanty** [comment]
@jfontan I did it here taking into account such things:
1) `regression-gitbase` already has a toolkit for preparation of server, repos etc
2) it takes ~30 seconds for this suite to pass, I did not want to include this heavy staff to `gitbase` repo and test it per commit or so I think this suite would fit better in a regression suite
3) I don't need to know the internals of `gitbase` so it's enough for me just to use it's binary here

Anyways I don't insist on storing these tests here, but in `gitbase` case it will take time to: 
1) rewrite the suite to standard golang tests form
2) implement preparations in `gitbase` that are already provided in `regression-gitbase`

File: bblfsh-mockups/services.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+   - timeout
+   - graph file
+   - ParseError is nil
+	node := nodes.String("a")
+	buf := bytes.NewBuffer(nil)
+	if err := nodesproto.WriteTo(buf, node); err != nil {
+		panic(err)
+	}
+
+	return &v2.ParseResponse{Language: "go", Uast: buf.Bytes(), Errors: []*v2.ParseError{{"wowwowwow"}}}, nil
```
Comment (ajnavarro) [inline bblfsh-mockups/services.go]: Leftover?

File: capture-output/capture_output.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	time.Sleep(delay)
+	if err := file.Close(); err != nil {
+		panic(err)
+	}
+
+	data, err := ioutil.ReadFile(file.Name())
+	if err != nil {
+		panic(err)
+	}
+	fmt.Println("data:", string(data))
```
Comment (ajnavarro) [inline capture-output/capture_output.go]: leftover?

**Review by ajnavarro — CHANGES_REQUESTED** [review]

**Review by erizocosmico — APPROVED** [review]
LGTM

File: bblfsh-mockups/services.go (hunk 1 above)
Comment (lwsanty) [reply bblfsh-mockups/services.go]: Done.

File: capture-output/capture_output.go (hunk 2 above)
Comment (lwsanty) [reply capture-output/capture_output.go]: Done.

**Review by jfontan — APPROVED** [review]

**Review by ajnavarro — APPROVED** [review]

**Pull request merged**